rhubarb/CMakeLists.txt

```diff
@@ -269,6 +269,38 @@ target_link_libraries(vorbis
 target_compile_options(vorbis PRIVATE ${disableWarningsFlags})
 set_target_properties(vorbis PROPERTIES FOLDER lib)
 
+# ... opus
+# lickily the complex build of libopus is managed by upstream itself.
+# The project is cmake based here. So it can support add_subdirectory().
+# the final static library is in the `opus*` subdir unfortunately.
+set(OPUS_BUILD_SHARED_LIBRARY OFF)
+set(OPUS_BUILD_TESTING OFF)
+set(OPUS_BUILD_PROGRAMS OFF)
+set(OPUS_CUSTOM_MODES ON)
+add_subdirectory(lib/opus-1.5.2)
```

Comment:
I didn't realize how big the Opus library is. I'm hesitant to add this much third-party code to this repo. Could you try using `FetchContent` instead? I used it just a few days ago for Google Test. Its documentation (like much of CMake) is atrocious, but the gist is this: If the referenced repo contains a top-level `CMakeLists.txt` file, it works much like `add_subdirectory`, but without adding adding the code to the repo. If it doesn't, it merely offers a convenient way to download files that can then be referenced like any others.

Reply:
@DanielSWolf  finally done, sorry for delay.

The patch has been rebased on your latest upstream changes.
Opus lib code is removed